salvo/ENVOY_DEVELOP_WORKFLOW.md

```diff
@@ -0,0 +1,135 @@
+
+    Finally you can also determine which Nighthawk version to use like Envoy's configuration, in this section we will build the latest Nighthawk version. You can find more details about configuration fields from [APIs definition](./api). 
+
+6. Details of test cases
+
+    The Benchmark Framework is developed via PyTest Framework. Therefore, the test cases file is defined as a python file contains the test cases. The [default test cases file](https://github.com/envoyproxy/nighthawk/blob/main/benchmarks/test/test_discovery.py) provides an example to configure Nighthawk client/server, outputs and test cases.
+
+    There are two main parts in the test case file.
+
+    - `_run_benchmark` function: 
+
+      At first, it defined a function named [`_run_benchmark`](https://github.com/envoyproxy/nighthawk/blob/main/benchmarks/test/test_discovery.py#L20) to run the specific PyTest fixture, which will define the behavior of Envoy and [Nighthawk test server](https://github.com/envoyproxy/nighthawk/blob/main/source/server/README.md) to be tested, you can find fixture definitions from these two files:
+
+      https://github.com/envoyproxy/nighthawk/blob/main/benchmarks/envoy_proxy.py
+
+      https://github.com/envoyproxy/nighthawk/blob/main/test/integration/integration_test_fixtures.py
+    
+      Then, the `_run_benchmark` function provides some configurable parameters used by Nighthawk client to send test requests, such as `rps`, `duration`, `request_body_size` and so on. By default, Nighthawk will use closed-loop mode, because closed-loop testing can be useful to get a sense of the maximum requests per second that can be squeezed out of a system (irrespective of the high latency incurred), as well as for comparing numbers with other load generators that use this methodology. More details about closed-loop mode: https://github.com/envoyproxy/nighthawk/blob/main/docs/root/terminology.md#closed-loop
+
+      Finally, `_run_benchmark` function defines the format of result outputs after the benchmark is done.
+
+    -  Test cases functions: 
+
+        After `_run_benchmark` function defined, you can write the specific test cases to use it. Take [`test_http_h1_small_request_small_reply_via`](https://github.com/envoyproxy/nighthawk/blob/8d6947b2d4b41edf0ea95a811d129c7a81629b86/benchmarks/test/test_discovery.py#L87) as an example, it will setup a Nighthawk client as the frontend, the Envoy Proxy to be tested as the load balancer, and a Nighthawk test server as the backend. The configurations of Envoy and test server can be assigned with Python decorators. Then you can invoke `_run_benchmark` function with specific fixture and parameters. 
+
+        More test cases examples: https://github.com/envoyproxy/nighthawk/blob/8d6947b2d4b41edf0ea95a811d129c7a81629b86/benchmarks/test/test_discovery.py#L92-#L119
+
+7. Run Salvo
+
+    To run Salvo, use the following command:
+
+    ```bash
+    bazel-bin/salvo --job <path to>/<binary_benchmark>.yaml
```

Comment:
(fyi) I wasn't able to execute this yet due to the examples containing non-existing commits. I will give it another try once we choose commits that exist in Envoy.

Reply:
Thanks, understood.